### PR TITLE
Ensure the app is always last in the middleware stack.

### PR DESCRIPTION
**Pull request by markstory**
By ensuring the app is always bound after the event, event listeners can be less careful. Without this plugins would always have to use `insertBefore()` which is kind of awkward.

### CODE REVIEW

File: src/Http/BaseApplication.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
         $cakeResponse = $this->getDispatcher()->dispatch($cakeRequest, $cakeResponse);
 
         // Convert the response back into a PSR7 object.
-        return $next($request, ResponseTransformer::toPsr($cakeResponse));
+        return ResponseTransformer::toPsr($cakeResponse);
```
Comment (lorenzo) [inline src/Http/BaseApplication.php]: What if someone decides to mount the app at another position? Should this still call `next()`?

Comment (markstory) [inline src/Http/BaseApplication.php]: How would they do that? The app is bound after all the middleware processing is done.

Comment (lorenzo) [inline src/Http/BaseApplication.php]: One of your examples in your talk is that you could test the app in isolation. Another reason could be composing 2 different apps.

Comment (markstory) [inline src/Http/BaseApplication.php]: Ok. I can revert this change then.

**Comment by markstory** [comment]
Does this look ok now @lorenzo ?

File: tests/TestCase/Http/ServerTest.php [hunk 2]
```diff
@@ -179,12 +179,19 @@ public function testBuildMiddlewareEvent()
...
+            $middleware->push(function ($req, $res, $next) {
+                $this->called = true;
+                return $next($req, $res);
+            });
+            $this->middleware = $middleware;
         });
         $server->run();
-        $this->assertTrue($called, 'Event not triggered.');
+        $this->assertTrue($this->called, 'Middleware added in the event was not triggered.');
+        $this->assertInstanceOf('Closure', $this->middleware->get(3), '2nd last middleware is a clousure');
```
Comment (Marlinc) [inline tests/TestCase/Http/ServerTest.php]: Typo in the comment, 'cloursure'.

Comment (markstory) [inline tests/TestCase/Http/ServerTest.php]: Thanks!

**Pull request merged**